### DISCUSSION TITLE
What does "Copilot Chat now knows about the active terminal's buffer and selection." mean for VScode IDE?

### DISCUSSION THREAD

**Discussion by muellerst-hg** [community/community#67083] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

The v1_81 release notes for the Github Copilot extension say:

> Terminal context
> Copilot Chat now knows about the active terminal's buffer and selection.
https://code.visualstudio.com/updates/v1_81#_terminal-context

What does "active terminal" refer to?

I wonder if Copilot reads and transmits content from my VScode shell opened with "Menu -> Terminal -> New Terminal"?

Could you please clearify?

**Comment by jzelAdmin2006** [comment, marked as answer]
Hi @muellerst-hg

The "active terminal" in VS Code refers to the currently open and focused terminal within the editor. When the v1_81 release note states that "Copilot Chat now knows about the active terminal's buffer and selection," it implies Copilot can access the content of the terminal you're working with, including its data and any text you've selected.

If you've opened a terminal in VS Code via "Menu -> Terminal -> New Terminal", that instance would be considered an "active terminal" when in focus. Thus, Copilot, based on this update, would be aware of its content.

![image](https://github.com/community/community/assets/112761658/8b77d400-b1ad-4935-bcd8-82f93ab1d9fc)
![image](https://github.com/community/community/assets/112761658/da3454d3-ef1a-42e5-a1ac-635d2626ce42)

For visual clarity, please refer to the two screenshots I've attached. They showcase the described behavior, useful if you don't have Copilot installed.

Best regards
jzelAdmin2006

**Reply by dawid-dahl-umain** [reply]
Hello! I just got Github Copilot Business and this doesn't work for me. Do you (or anyone) know if I need to turn on some Beta flag or something? Thank you!

**Reply by georgemac-labs** [reply]
Not working for me either

**Comment by muellerst-hg** [comment]
So as a consequence, it sounds to me that any credentials entered into the terminal would potentially be submitted to Github/OpenAI (other peoples computer).

**Reply by jzelAdmin2006** [reply]
Hi @muellerst-hg

Sorry for the waiting time. I'm quite busy sometimes.

Yes, that is correct and a problem if you are dealing with truly sensitive credentials (e.g. in a local environment it wouldn't actually matter). In the case of OpenAI, it is difficult to circumvent the problem because they only offer their AI model via API. If privacy is important to you and you still want to have an AI pair programmer, you would theoretically have to use a model that is available for download. It's possible that something like this will be available in the future, but at the moment I don't know of any viable alternative.

Best regards
jzelAdmin2006

**Reply by muellerst-hg** [reply]
[Tabby: Self-hosted AI coding assistant](https://github.com/TabbyML/tabby)
